scripts/jenkins/Jenkinsfile.release

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env groovy
+
+// Imports
+import java.text.SimpleDateFormat
+
+// Buildjob properties
+properties([
+  buildDiscarder(
+    logRotator(
+      artifactDaysToKeepStr: '31',  // Keep artifacts for max 31 days
+      artifactNumToKeepStr: '10'    // Keep artifacts for last 5 builds
+    )
+  )
+])
+
+// Globals
+DOCKER_NODE_LABEL = 'docker'
+REGISTRY = 'hub.libelektra.org'
+
+// Define DOCKER_OPTS enum which is used in withDockerEnv
+enum DOCKER_OPTS {
+  MOUNT_MIRROR,       // Mount the git repository mirror into the container
+  PTRACE              // Allow ptraces in the container
+  public DOCKER_OPTS()  {} // WORKAROUND see TEST
+}
+
+DOCKER_IMAGES = [:]  // Containers docker image descriptions, populated during
+                     // dockerInit()
+
+
+
+// Stages
+stage("Init docker images") {
+  dockerInit()
+}
+
+stage("Pull docker images") {
+  parallel generateDockerPullStages()
+}
+
+maybeStage("Build docker images", DOCKER_IMAGES.any {img -> !img.value.exists}) {
+  lock('docker-images') {
+        parallel generateDockerBuildStages()
+    }
+}
+
+stage("Release") {
+    milestone label: "Release"
+    buildRelease()
+}
+
+/**************/
+/* Generators */
+/**************/
+
+/* Populate DOCKER_IMAGES with data
+ *
+ * For this we need a checkout of the scm to generate the hash for the
+ * Dockerfiles which indicates if a rebuild of the images is needed
+ */
+def dockerInit() {
```

Comment:
I am afraid the answer will be no, but is it possible to avoid the duplication between the Jenkinsfile and have a common part which defines all the docker images?

Btw. it can be done after this release.

Reply:
Yes, we could extract the common stages/functions into a Jenkins Shared Library.